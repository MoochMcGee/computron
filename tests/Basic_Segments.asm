[bits 16]

mov ax, 0x9abc
mov es, ax
mov ax, 0xdef0
mov ss, ax

mov ax, 0x1234
mov ds, ax

; Plant suicide instruction at 1234:0000
mov [0], byte 0xf1

mov ax, 0x5678
mov ds, ax

jmp 0x1234:0000
