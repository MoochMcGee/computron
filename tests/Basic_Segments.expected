1000:00000000 B8 EAX=00000000 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000003 8E EAX=00009ABC EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000005 B8 EAX=00009ABC EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=9ABC SS=9000 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000008 8E EAX=0000DEF0 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=9ABC SS=9000 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:0000000A B8 EAX=0000DEF0 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:0000000D 8E EAX=00001234 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:0000000F C6 EAX=00001234 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1234 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000014 B8 EAX=00001234 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1234 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000017 8E EAX=00005678 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1234 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000019 EA EAX=00005678 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=5678 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1234:00000000 F1 EAX=00005678 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=5678 ES=9ABC SS=DEF0 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
